app/code/Magento/Translation/view/base/templates/translate.phtml

```diff
@@ -26,7 +26,7 @@
                 $.initNamespaceStorage('mage-translation-file-version');
                 versionObj = $.localStorage.get('mage-translation-file-version');
 
-                <?php $version = sha1($block->getTranslationFileTimestamp() . $block->getTranslationFilePath()); ?>
+                <?php $version = sha1(sha1_file($block->getTranslationFileFullPath()) . $block->getTranslationFilePath()); ?>
```

Comment:
Hashes calculation and operations on the filesystem should not be a responsibility of the presentation layer.